src/pgduckdb_types.cpp

```diff
@@ -192,6 +169,269 @@ ConvertNumeric(T value, idx_t scale) {
 	return result;
 }
 
+static Datum
+ConvertNumericDatum(const duckdb::Value &value) {
+	if (value.type().id() == duckdb::LogicalTypeId::DOUBLE) {
```

Comment:
nit: we could assign `value.type().id()` to a const here and re-use it a lot in this function